### PR TITLE
Make keep alive optional in PointInTimeBuilder

### PR DESCRIPTION
**Pull request by dnhatn**
Remove the keepAlive parameter from the constructor of PointInTimeBuilder as it's optional.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java [hunk 1]
```diff
@@ -1702,7 +1701,12 @@ public static PointInTimeBuilder fromXContent(XContentParser parser) throws IOEx
             if (params.id == null) {
                 throw new IllegalArgumentException("point int time id is not provided");
             }
-            return new PointInTimeBuilder(params.id, params.keepAlive);
+            return new PointInTimeBuilder(params.id).setKeepAlive(params.keepAlive);
+        }
+
+        public PointInTimeBuilder setKeepAlive(TimeValue keepAlive) {
```
Comment (jimczi) [inline server/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java:1707]: Can you add javadoc

File: server/src/main/java/org/elasticsearch/action/search/SearchRequestBuilder.java [hunk 2]
```diff
@@ -549,7 +549,7 @@ public SearchRequestBuilder setCollapse(CollapseBuilder collapse) {
      * @param keepAlive       the extended time to live for the search context
      */
     public SearchRequestBuilder setSearchContext(String searchContextId, TimeValue keepAlive) {
```
Comment (jimczi) [inline server/src/main/java/org/elasticsearch/action/search/SearchRequestBuilder.java:551]: Should this be renamed to setPointInTime or setPointInTimeBuilder ?

**Review by jimczi — APPROVED** [review]
I left some comments but the change makes sense to me.

**Comment by dnhatn** [comment]
@elasticmachine test this please

File: server/src/main/java/org/elasticsearch/action/search/SearchRequestBuilder.java (hunk 2 above)
Comment (dnhatn) [reply server/src/main/java/org/elasticsearch/action/search/SearchRequestBuilder.java]: ++. I pushed https://github.com/elastic/elasticsearch/pull/62720/commits/5af72036b99e9f0c78f116948404f67af036fbe9

File: server/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java (hunk 1 above)
Comment (dnhatn) [reply server/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java]: Added in https://github.com/elastic/elasticsearch/pull/62720/commits/5af72036b99e9f0c78f116948404f67af036fbe9

**Comment by dnhatn** [comment]
run elasticsearch-ci/packaging-sample-windows

**Pull request merged**